Backend/src/main/java/com/example/SkyLine/service/NotificationService.java

```diff
@@ -1,70 +1,249 @@
 import com.example.SkyLine.repository.NotificationRepository;
 import com.example.SkyLine.repository.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+import java.time.ZoneId;
 import java.util.ArrayList;
+import java.util.List;
 
 @Service
 public class NotificationService {
     private final NotificationRepository notificationRepository;
     private final UserRepository userRepository;
+    private final CompanyRepository companyRepository;
+
+    private final ClientRepository clientRepository;
     @Autowired
-    public NotificationService(NotificationRepository notificationRepository, UserRepository userRepository) {
+    public NotificationService(NotificationRepository notificationRepository, UserRepository userRepository, CompanyRepository companyRepository, ClientRepository clientRepository) {
         this.notificationRepository = notificationRepository;
         this.userRepository = userRepository;
+        this.companyRepository = companyRepository;
+        this.clientRepository = clientRepository;
+
     }
     public boolean notify(int requesterID, int candidateID){
         long previousRequests=notificationRepository.countCandidatePreviousRequests(candidateID);
+        System.out.println("previousRequests= "+previousRequests);
         if(previousRequests>0){return false;}
-        ArrayList<Integer> SuperAdminsIDs=userRepository.getSuperAdminsID("SuperAdmin");
+        ArrayList<Integer> SuperAdminsIDs=userRepository.getSuperAdminsID("SUPERADMIN");
+        System.out.println("SuperAdminsIDs= "+SuperAdminsIDs.size());
         for (Integer SuperAdminsID : SuperAdminsIDs
              ) {
             Notification request=new Notification(requesterID,candidateID,SuperAdminsID);
             notificationRepository.save(request);
         }
 
-
-
         return true;
 
         
     }
- public void approveRequest(Notification request){
-        notificationRepository.deleteNotificationById(request.getRequester_id(), request.getCandidate_id(), request.getResponder_id());
-        notificationRepository.updateNotificationById(request.getRequester_id(), request.getCandidate_id(), true, new java.sql.Date(System.currentTimeMillis()) );
+
+    public boolean request(int requesterId, int candidateId){
+        UserRoleEnum role=userRepository.findUserById(requesterId).getUserRole();
+        if (role==UserRoleEnum.ADMIN){
+            System.out.println("here ");
+           return notify(requesterId,candidateId);
+        }
+        else if(role==UserRoleEnum.COMPANY){
+           Notification request=new Notification(requesterId,candidateId);
+              notificationRepository.save(request);
+              return true;
+        }
+        else {
+           System.out.println("You are not allowed to request ");
+           return false;
+        }
+
+
+    }
+ public void approveRequest(int NotificationId) {
```

Comment:
I think you should use @Transactional for approve/reject requests and combine them in a single function so when multiple admins/company handle these requests its consistent